sdk/scheduler/src/main/java/com/mesosphere/sdk/curator/CuratorPersister.java

```diff
@@ -198,6 +198,94 @@ public static Builder newBuilder(String serviceName, String zookeeperHostPort) {
+        final String path = withFrameworkPrefix(unprefixedPath);
+        LOGGER.debug("Setting {} => {}", path, getInfo(bytes));
+        try {
+            try {
+                client.create().creatingParentsIfNeeded().forPath(path, bytes);
+            } catch (KeeperException.NodeExistsException e) {
+                client.setData().forPath(path, bytes);
+            }
+        } catch (Exception e) {
+            throw new PersisterException(Reason.STORAGE_ERROR,
+                    String.format("Unable to set %d bytes in %s", bytes.length, path), e);
+        }
+    }
+
+    @Override
+    public Map<String, byte[]> getMany(Collection<String> unprefixedPaths) throws PersisterException {
+        if (unprefixedPaths.isEmpty()) {
+            return Collections.emptyMap();
+        }
+        LOGGER.debug("Getting {} entries: {}", unprefixedPaths.size(), unprefixedPaths);
+
+        Map<String, byte[]> result = new TreeMap<>();
+        // Unlike with writes, there is not an atomic read operation. Therefore we wing it with a series of plain reads.
+        // We could conceivably add some form of locking here to avoid e.g. a race with another thread doing writes at
+        // the same time, but assuming the PersisterCache is enabled, this function wouldn't be getting called anyway,
+        // as the PersisterCache would have fetched all the data up-front to be served from memory. If this assumption
+        // changes, then it may make sense to look into some form of proper read locking here.
+        for (String unprefixedPath : unprefixedPaths) {
+            String path = withFrameworkPrefix(unprefixedPath);
+            try {
+                result.put(unprefixedPath, client.getData().forPath(path));
+            } catch (KeeperException.NoNodeException e) {
+                result.put(unprefixedPath, null);
+            } catch (Exception e) {
+                throw new PersisterException(Reason.STORAGE_ERROR,
+                        String.format("Unable to retrieve data from %s", path), e);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public void setMany(Map<String, byte[]> unprefixedPathBytesMap) throws PersisterException {
+        if (unprefixedPathBytesMap.isEmpty()) {
+            return;
+        }
+        // Convert map to translated prefixed paths. Manually construct TreeMap for consistent ordering (for tests):
+        Map<String, byte[]> pathBytesMap = new TreeMap<>();
+        for (Map.Entry<String, byte[]> entry : unprefixedPathBytesMap.entrySet()) {
+            pathBytesMap.put(withFrameworkPrefix(entry.getKey()), entry.getValue());
+        }
+        LOGGER.debug("Updating {} entries: {}", pathBytesMap.size(), pathBytesMap.keySet());
+        runTransactionWithRetries(new SetTransactionFactory(pathBytesMap));
+    }
+
+    @Override
+    public void recursiveCopy(String src, String dest) throws PersisterException {
```

Comment:
It'd probably make sense to follow the pattern elsewhere in CuratorPersister: name these params `unprefixedSrc`/`unprefixedDest`, then start the function with `String src=withFrameworkPrefix(unprefixedSrc);`

The main reason for doing this is to avoid accidentally using the unprefixed version of a path when performing the copy.